### PR TITLE
[Dawn] Component Code Cleanup

### PR DESCRIPTION
**Pull request by kevinmasd**
Going to be one in a series to cleanup code before the semester starts

* The old Container Model is switched into the Component model for naming, since its the wrapper we are specifically referencing
  * Files that separate the two are merged together, with exception of Editor due to size and NameEdit due to questionable future
* Initial Changes to add [PropTypes](https://github.com/facebook/prop-types)
* Functions made simpler
  * i.e. No reason to make function if functionality fits in one line, not used elsewhere
* Initial Props cleanup (Haven't removed branches due to unknown efficiency changes in tree reconciliation)

### CODE REVIEW

File: dawn/package.json [hunk 1]
```diff
@@ -50,8 +50,9 @@
   },
   "dependencies": {
     "brace": "0.10.0",
-    "circular-json": "0.3.1",
-    "create-react-class": "15.6.0",
+    "bufferutil": "3.0.2",
+    "circular-json": "^0.3.1",
+    "create-react-class": "^15.6.0",
```
Comment (kevinji) [inline dawn/package.json]: These two should be exactly-versioned.

File: dawn/webpack.config.babel.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ const plugins = [
   new webpack.DefinePlugin({
     VERSION: JSON.stringify(version),
   }),
+  new webpack.IgnorePlugin(/vertx/),
```
Comment (kevinji) [inline dawn/webpack.config.babel.js]: Is there a reason this is needed?

File: dawn/renderer/components/peripherals/Motor.js [hunk 3]
```diff
@@ -1,37 +1,36 @@
...
-import { ProgressBar } from 'react-bootstrap';
 import numeral from 'numeral';
 
+import { ProgressBar } from 'react-bootstrap';
+
 /**
- * A component representing a motor.
+ *  Motor Component
  */
 const Motor = props => (
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/Motor.js]: Instead of `props`, you can probably just spread out the attributes: `{ id, param, device_name }`.

File: dawn/renderer/components/peripherals/GenericPeripheral.js [hunk 4]
```diff
@@ -1,36 +1,34 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import _ from 'lodash';
 import numeral from 'numeral';
 
 /**
- * A generic peripheral, used when the peripheralType is unknown.
+ * Generic Peripheral for General Case
  */
 const GenericPeripheral = props => (
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/GenericPeripheral.js]: Instead of `props`, you can probably just spread out the attributes: `{ id, param, device_name }`.

File: dawn/renderer/components/peripherals/BooleanSensor.js [hunk 5]
```diff
@@ -1,51 +1,45 @@
...
+import PropTypes from 'prop-types';
 import _ from 'lodash';
 import { PeripheralTypes } from '../../constants/Constants';
 
 /**
- * A component representing a boolean sensor,
- * for example a LimitSwitch.
+ * Boolean Sensor Component
  */
 class BooleanSensor extends React.Component {
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/BooleanSensor.js]: Since this has no state, can you convert this to a functional-style component, and convert `formatBoolean` to a helper function?

File: dawn/renderer/components/StatusLabel.js [hunk 6]
```diff
@@ -14,19 +16,29 @@ const StatusLabel = (props) => {
...
+const mapStateToProps = state => ({
+  batteryLevel: state.peripherals.batteryLevel,
+  batterySafety: state.peripherals.batterySafety,
+});
+
+
+const StatusLabel = connect(mapStateToProps)(StatusLabelComponent);
+
 export default StatusLabel;
+
```
Comment (kevinji) [inline dawn/renderer/components/StatusLabel.js]: Stray newline.

File: dawn/renderer/components/StatusLabel.js [hunk 7]
```diff
@@ -14,19 +16,29 @@ const StatusLabel = (props) => {
...
+  batterySafety: PropTypes.bool.isRequired,
 };
 
+
+const mapStateToProps = state => ({
+  batteryLevel: state.peripherals.batteryLevel,
+  batterySafety: state.peripherals.batterySafety,
+});
+
+
```
Comment (kevinji) [inline dawn/renderer/components/StatusLabel.js]: Stray newline.

File: dawn/package.json [hunk 8]
```diff
@@ -76,6 +78,7 @@
     "smalltalk": "2.4.1",
     "ssh2": "0.5.5",
     "superagent": "3.5.2",
+    "utf-8-validate": "3.0.3",
```
Comment (kevinji) [inline dawn/package.json:81]: Is this used anywhere?

File: dawn/package.json [hunk 9]
```diff
@@ -50,8 +50,9 @@
   },
   "dependencies": {
     "brace": "0.10.0",
-    "circular-json": "0.3.1",
-    "create-react-class": "15.6.0",
+    "bufferutil": "3.0.2",
```
Comment (kevinji) [inline dawn/package.json]: Is this used anywhere?

**Review by kevinji — CHANGES_REQUESTED** [review]
Seems mostly reasonable. There are a bunch of places that use `PropTypes` without making them required; it'd be nice if you did a full pass and made the actually required props `isRequired`.

File: dawn/package.json (hunk 8 above)
Comment (kevinmasd) [reply dawn/package.json:81]: Recommended dependency for ws.

File: dawn/package.json (hunk 9 above)
Comment (kevinmasd) [reply dawn/package.json]: Recommended dependency for ws.

File: dawn/package.json (hunk 1 above)
Comment (kevinmasd) [reply dawn/package.json]: Will change

File: dawn/webpack.config.babel.js (hunk 2 above)
Comment (kevinmasd) [reply dawn/webpack.config.babel.js]: Sometimes there's a weird bug when `make watch` stops due to finding a missing item. This was an attempt to prevent that, but it seems there are other bugs.

Comment (kevinmasd) [reply dawn/webpack.config.babel.js]: Specifically I get an issue similar to https://github.com/redux-saga/redux-saga/issues/197. I'll debug on a separate branch.

File: dawn/renderer/components/peripherals/BooleanSensor.js [hunk 10]
```diff
@@ -1,51 +1,42 @@
+/* eslint-disable camelcase */
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/BooleanSensor.js:1]: Why is this necessary?

File: dawn/renderer/components/peripherals/BooleanSensor.js [hunk 11]
```diff
@@ -1,51 +1,42 @@
...
-    if (peripheralType === PeripheralTypes.LimitSwitch) {
-      return (sensorValue) ? 'Closed' : 'Open';
-    }
-    return sensorValue;
+const formatBoolean = (peripheralType, sensorValue) => {
+  if (peripheralType === PeripheralTypes.LimitSwitch) {
+    return (sensorValue) ? 'Closed' : 'Open';
   }
+  return sensorValue; // TODO: Verify if int_value or bool_value
+};
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/BooleanSensor.js:15]: This function seems a bit strange; the `if` condition expects a boolean (or truthy `sensorValue`) and returns a string, while the `else` branch expects an int (I think)?

File: dawn/renderer/components/peripherals/GenericPeripheral.js [hunk 12]
```diff
@@ -1,36 +1,35 @@
+/* eslint-disable camelcase */
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/GenericPeripheral.js:1]: Same comment as other file: why is this necessary?

File: dawn/renderer/components/peripherals/Motor.js [hunk 13]
```diff
@@ -1,37 +1,37 @@
+/* eslint-disable camelcase */
```
Comment (kevinji) [inline dawn/renderer/components/peripherals/Motor.js:1]: Ditto.

File: dawn/webpack.config.babel.js (hunk 2 above)
Comment (kevinji) [reply dawn/webpack.config.babel.js]: Can you leave a comment here mentioning that this is an issue?

**Review by kevinji — CHANGES_REQUESTED** [review]
Just a few more minor changes.

File: dawn/renderer/components/peripherals/BooleanSensor.js (hunk 10 above)
Comment (kevinmasd) [reply dawn/renderer/components/peripherals/BooleanSensor.js:1]: It flags on the `device_name`, `device_type`, etc variables when unpacking from the object.

File: dawn/renderer/components/peripherals/BooleanSensor.js (hunk 11 above)
Comment (kevinmasd) [reply dawn/renderer/components/peripherals/BooleanSensor.js:15]: `sensorValue` as far as I can tell is always numerical. It just happens that for one sensor, we have semantics that allow us to translate value to word meaning.

**Comment by kevinmasd** [comment]
The vertx thing is going to be removed, since it seems to hit a different bug preventing recompilation.

**Review by kev-chien — APPROVED** [review]
Looks good to me.

**Review by kevinji — APPROVED** [review]
